### ISSUE TITLE
Opening a matplotlib figure, causes kivy desktop app to shrink (size & font)

### ISSUE THREAD

**Issue by abedhammoud** [kivy/kivy#7094]
**Software Versions**
* Python: 3.8.5
* OS: win10
* Kivy: 2.0.0rc3
* Kivy installation method: pip

**Describe the bug**
When I call any of the matplotlib plotting routines, as the matplotlib figure is displayed, the Kivy desktop application shrinks in size (window size and font)

**Expected behavior**
The display of the matplotlib should not affect the size/font of the Kivy application.

**To Reproduce**
Please find the code below.

**Code and Logs and screenshots**
```python
import kivy
from kivy.app import App
from kivy.uix.boxlayout import BoxLayout

import matplotlib.pyplot as plt

kivy.lang.Builder.load_string('''

<DataView>:

    orientation: 'vertical'
    spacing: 5
    padding: 5

    Label:
        text: 'HELLO WORLD'
''', filename='dataview.kv')

class DataView(BoxLayout):

    def __init__(self, **kwargs):
        super().__init__(**kwargs)

class TestApp(App):

    def build(self):
        DV = DataView()

        # Commenting/uncommenting this line causes the kivy application
        # font size / window size to change.
        fig, ax = plt.subplots(2, 2)

        return DV

if __name__ == '__main__':
    TestApp().run()
```

**Additional context**

Kivy debugging dump:

[INFO   ] [deps        ] Successfully imported "kivy_deps.gstreamer" 0.2.0
[INFO   ] [deps        ] Successfully imported "kivy_deps.glew" 0.2.0
[INFO   ] [deps        ] Successfully imported "kivy_deps.sdl2" 0.2.0
[INFO   ] [Kivy        ] v2.0.0rc3, git-20c14b2, 20200615
[INFO   ] [Kivy        ] Installed at "C:\Users\abed\anaconda3\lib\site-packages\kivy\__init__.py"
[INFO   ] [Python      ] v3.8.5 | packaged by conda-forge | (default, Sep 16 2020, 17:19:16) [MSC v.1916 64 bit (AMD64)]
[INFO   ] [Python      ] Interpreter at "C:\Users\abed\anaconda3\python.exe"
[INFO   ] [Factory     ] 185 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil (img_ffpyplayer, img_gif ignored)
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] GLEW initialization succeeded
[INFO   ] [GL          ] Backend used <glew>
[INFO   ] [GL          ] OpenGL version <b'4.6.0 - Build 26.20.100.7637'>
[INFO   ] [GL          ] OpenGL vendor <b'Intel'>
[INFO   ] [GL          ] OpenGL renderer <b'Intel(R) UHD Graphics 620'>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <b'4.60 - Build 26.20.100.7637'>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Text        ] Provider: sdl2
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available

**Comment by MaxHausmann** [comment]
Hello,

I have the same issue. I changed from garden.graph to matplotlib.
Since then the content of the main app is very small at startup.
Resizing the window solves the problem but for me it is not a suitable solution.

Got no errors or something similar.

**Comment by abedhammoud** [comment]
Hi @MaxHausmann ,

yes, I still see the issue with Kivy v2.0.0rc4, python 3.8.6, and matplotlib version 3.3.3

In fact, I also noticed that the same effect happens when my application starts pandas profiling in the browser.

With matplotlib, I tried setting mpl.rcParams['figure.dpi'] to different DPI (from 60 to 300), and I continued to see the same effect.
Mainly that my main application window shrinks in size and font when a figure is displayed in a separate matplotlib window or in the browser.

**Comment by tshirtman** [comment]
I would suggest binding a function to the Window.size property, and putting a debugger in that function. I tried doing it, but it didn't hit here in linux, that might be windows specific.

```
import kivy
from kivy.app import App
from kivy.core.window import Window
from kivy.factory import Factory as F

kivy.lang.Builder.load_string('''
#:import plt matplotlib.pyplot
<DataView@BoxLayout>:
    orientation: 'vertical'
    spacing: 5
    padding: 5

    Label:
        text: 'HELLO WORLD'

    Button:
        text: 'hit me!'
        on_release: fig, ax = plt.subplots(2, 2)
''', filename='dataview.kv')


class TestApp(App):
    def build(self):
        Window.bind(size=self.debug)
        DV = F.DataView()

        # Commenting/uncommenting this line causes the kivy application
        # font size / window size to change.

        return DV

    def debug(self, *args):
        __import__('pdb').set_trace()


if __name__ == '__main__':
    TestApp().run()
```
and after pressing the button, the debugger should be triggered type `u` then enter, and repeat until arriving at code outside of kivy that triggers the window's resizing.

**Comment by abedhammoud** [comment]
Thanks, @tshirtman, good idea.

I ran the above code, I get the normal screen/font size showing, then when I click the button, and as expected, the window/font size shrinks. 

What is not expected, is that the debug method is never called, ( I also put a print inside). This implies that even though the window size is getting smaller, the actual window width and height (size) is not really changing - just the DPI I assume?

Thanks for any suggestion, I am happy to try more tests.

**Comment by abedhammoud** [comment]
Just a quick update in case it can be useful to anyone.

Confirming a comment from @tshirtman (thank you), I installed the application on ubuntu 20.10, and all worked as expected. I don't see the behavior explained above.

I went back to windows 10, installed anaconda, then with conda I installed kivy 2.0:  the issue persists.

So it seems the problem is specific to Windows 10. Basically, anytime I open a matplotlib figure, my kivy app shrinks in size (without triggering any size change callback).

In both cases, I am running kivy 2.0, with python 3.8.6

Best wishes to all for the holidays - and a big thank you to the Kivy dev team - one year since I started using kivy and I like it very much.

**Comment by matham** [comment]
If I were to guess, I'd guess that the matplotlib backend provider is calling into Windows to change the DPI awareness or something like that. You can see a somewhat related discussion https://github.com/kivy/kivy/issues/3705 about the effects of doing this.

I would try running matplotlib with different backends to see if it changes. Alternatively, change Kivy's density at startup (like in the example code there) and see if it looks ok after importing matplotlib.

**Comment by abedhammoud** [comment]
Thanks, @matham for the pointer. I have not seen #3705 before. I will dig deeper and see what I can find/learn.

**Comment by probablyfine** [comment]
I tried to replicate this problem but surprisingly I can't. I'm on Windows 10 (version 2004, build 19041.685), with kivy 2.0.0, matplotlib 3.3.3, and Python 3.7.1.

@abedhammoud, can you try to maximize the kivy window and re-try @tshirtman's suggestion to debug on the Window.size trigger? If the window is maximized, Window.size should change with the DPI change; otherwise it won't because the window shrinks as well. You'll get stuck in the debugger when you maximize the window, so just continue after that and open a matplotlib figure as before.

**Comment by tshirtman** [comment]
@probablyfine is your windows display zoom setting set to 100%? i think the issue is only when it's to a different value.

**Comment by abedhammoud** [comment]
Thank you both.

@probablyfine  I tried your recommendation on two win10 laptops and I still get the same effect. I am running Win10 Pro version 20H2, build 19042.685 with Kivy 2.0, matplotlib 3.3.3, and python  3.8.6 (and a zoom of 150%). Note that even with the window maximized, no Window.size event is fired when the window shrinks (after I click the button).

Following the comments of @tshirtman, I changed the zoom to 100%, and the shrinking stopped. I have not noticed that before, thanks @tshirtman.

thank you both for your help, really appreciated

Here is the log - I don't see anything special.

```python
[INFO   ] [Kivy        ] v2.0.0
[INFO   ] [Kivy        ] Installed at "C:\Users\abed\anaconda3\lib\site-packages\kivy\__init__.py"
[INFO   ] [Python      ] v3.8.6 | packaged by conda-forge | (default, Dec 26 2020, 04:30:06) [MSC v.1916 64 bit (AMD64)]
[INFO   ] [Python      ] Interpreter at "C:\Users\abed\anaconda3\python.exe"
[INFO   ] [Factory     ] 186 symbols loaded
[DEBUG  ] [Cache       ] register <kv.lang> with limit=None, timeout=None
[DEBUG  ] [Cache       ] register <kv.image> with limit=None, timeout=60
[DEBUG  ] [Cache       ] register <kv.atlas> with limit=None, timeout=None
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil (img_ffpyplayer ignored)
[DEBUG  ] [Cache       ] register <kv.texture> with limit=1000, timeout=60
[DEBUG  ] [Cache       ] register <kv.shader> with limit=1000, timeout=3600
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] GLEW initialization succeeded
[DEBUG  ] [GL          ] available extensions: b'GL_3DFX_texture_compression_FXT1 GL_AMD_depth_clamp_separate GL_AMD_vertex_shader_layer GL_AMD_vertex_shader_viewport_index GL_ARB_ES2_compatibility GL_ARB_ES3_1_compatibility GL_ARB_ES3_compatibility GL_ARB_arrays_of_arrays GL_ARB_base_instance GL_ARB_bindless_texture GL_ARB_blend_func_extended GL_ARB_buffer_storage GL_ARB_cl_event GL_ARB_clear_buffer_object GL_ARB_clear_texture GL_ARB_clip_control GL_ARB_color_buffer_float GL_ARB_compatibility GL_ARB_compressed_texture_pixel_storage GL_ARB_compute_shader GL_ARB_conditional_render_inverted GL_ARB_conservative_depth GL_ARB_copy_buffer GL_ARB_copy_image GL_ARB_cull_distance GL_ARB_debug_output GL_ARB_depth_buffer_float GL_ARB_depth_clamp GL_ARB_depth_texture GL_ARB_derivative_control GL_ARB_direct_state_access GL_ARB_draw_buffers GL_ARB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_draw_indirect GL_ARB_draw_instanced GL_ARB_enhanced_layouts GL_ARB_explicit_attrib_location GL_ARB_explicit_uniform_location GL_ARB_fragment_coord_conventions GL_ARB_fragment_layer_viewport GL_ARB_fragment_program GL_ARB_fragment_program_shadow GL_ARB_fragment_shader GL_ARB_fragment_shader_interlock GL_ARB_framebuffer_no_attachments GL_ARB_framebuffer_object GL_ARB_framebuffer_sRGB GL_ARB_geometry_shader4 GL_ARB_get_program_binary GL_ARB_get_texture_sub_image GL_ARB_gl_spirv GL_ARB_gpu_shader5 GL_ARB_gpu_shader_fp64 GL_ARB_half_float_pixel GL_ARB_half_float_vertex GL_ARB_indirect_parameters GL_ARB_instanced_arrays GL_ARB_internalformat_query GL_ARB_internalformat_query2 GL_ARB_invalidate_subdata GL_ARB_map_buffer_alignment GL_ARB_map_buffer_range GL_ARB_multi_bind GL_ARB_multi_draw_indirect GL_ARB_multisample GL_ARB_multitexture GL_ARB_occlusion_query GL_ARB_occlusion_query2 GL_ARB_pipeline_statistics_query GL_ARB_pixel_buffer_object GL_ARB_point_parameters GL_ARB_point_sprite GL_ARB_polygon_offset_clamp GL_ARB_post_depth_coverage GL_ARB_program_interface_query GL_ARB_provoking_vertex GL_ARB_query_buffer_object GL_ARB_robust_buffer_access_behavior GL_ARB_robustness GL_ARB_robustness_isolation GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_seamless_cubemap_per_texture GL_ARB_separate_shader_objects GL_ARB_shader_atomic_counter_ops GL_ARB_shader_atomic_counters GL_ARB_shader_bit_encoding GL_ARB_shader_draw_parameters GL_ARB_shader_group_vote GL_ARB_shader_image_load_store GL_ARB_shader_image_size GL_ARB_shader_objects GL_ARB_shader_precision GL_ARB_shader_stencil_export GL_ARB_shader_storage_buffer_object GL_ARB_shader_subroutine GL_ARB_shader_texture_image_samples GL_ARB_shading_language_100 GL_ARB_shading_language_420pack GL_ARB_shading_language_packing GL_ARB_shadow GL_ARB_spirv_extensions GL_ARB_stencil_texturing GL_ARB_sync GL_ARB_tessellation_shader GL_ARB_texture_barrier GL_ARB_texture_border_clamp GL_ARB_texture_buffer_object GL_ARB_texture_buffer_object_rgb32 GL_ARB_texture_buffer_range GL_ARB_texture_compression GL_ARB_texture_compression_bptc GL_ARB_texture_compression_rgtc GL_ARB_texture_cube_map GL_ARB_texture_cube_map_array GL_ARB_texture_env_add GL_ARB_texture_env_combine GL_ARB_texture_env_crossbar GL_ARB_texture_env_dot3 GL_ARB_texture_filter_anisotropic GL_ARB_texture_float GL_ARB_texture_gather GL_ARB_texture_mirror_clamp_to_edge GL_ARB_texture_mirrored_repeat GL_ARB_texture_multisample GL_ARB_texture_non_power_of_two GL_ARB_texture_query_levels GL_ARB_texture_query_lod GL_ARB_texture_rectangle GL_ARB_texture_rg GL_ARB_texture_rgb10_a2ui GL_ARB_texture_stencil8 GL_ARB_texture_storage GL_ARB_texture_storage_multisample GL_ARB_texture_swizzle GL_ARB_texture_view GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_ARB_transform_feedback_instanced GL_ARB_transform_feedback_overflow_query GL_ARB_transpose_matrix GL_ARB_uniform_buffer_object GL_ARB_vertex_array_bgra GL_ARB_vertex_array_object GL_ARB_vertex_attrib_64bit GL_ARB_vertex_attrib_binding GL_ARB_vertex_buffer_object GL_ARB_vertex_program GL_ARB_vertex_shader GL_ARB_vertex_type_10f_11f_11f_rev GL_ARB_vertex_type_2_10_10_10_rev GL_ARB_viewport_array GL_ARB_window_pos GL_ATI_separate_stencil GL_EXT_abgr GL_EXT_bgra GL_EXT_blend_color GL_EXT_blend_equation_separate GL_EXT_blend_func_separate GL_EXT_blend_minmax GL_EXT_blend_subtract GL_EXT_clip_volume_hint GL_EXT_compiled_vertex_array GL_EXT_direct_state_access GL_EXT_draw_buffers2 GL_EXT_draw_range_elements GL_EXT_fog_coord GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_framebuffer_object GL_EXT_geometry_shader4 GL_EXT_gpu_program_parameters GL_EXT_gpu_shader4 GL_EXT_multi_draw_arrays GL_EXT_packed_depth_stencil GL_EXT_packed_float GL_EXT_packed_pixels GL_EXT_polygon_offset_clamp GL_EXT_rescale_normal GL_EXT_secondary_color GL_EXT_separate_specular_color GL_EXT_shader_framebuffer_fetch GL_EXT_shader_integer_mix GL_EXT_shadow_funcs GL_EXT_stencil_two_side GL_EXT_stencil_wrap GL_EXT_texture3D GL_EXT_texture_array GL_EXT_texture_compression_s3tc GL_EXT_texture_edge_clamp GL_EXT_texture_env_add GL_EXT_texture_env_combine GL_EXT_texture_filter_anisotropic GL_EXT_texture_integer GL_EXT_texture_lod_bias GL_EXT_texture_rectangle GL_EXT_texture_sRGB GL_EXT_texture_sRGB_decode GL_EXT_texture_shared_exponent GL_EXT_texture_snorm GL_EXT_texture_storage GL_EXT_texture_swizzle GL_EXT_timer_query GL_EXT_transform_feedback GL_IBM_texture_mirrored_repeat GL_INTEL_conservative_rasterization GL_INTEL_fragment_shader_ordering GL_INTEL_framebuffer_CMAA GL_INTEL_map_texture GL_INTEL_multi_rate_fragment_shader GL_INTEL_performance_query GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_KHR_context_flush_control GL_KHR_debug GL_KHR_no_error GL_KHR_texture_compression_astc_hdr GL_KHR_texture_compression_astc_ldr GL_NV_blend_square GL_NV_conditional_render GL_NV_primitive_restart GL_NV_texgen_reflection GL_SGIS_generate_mipmap GL_SGIS_texture_edge_clamp GL_SGIS_texture_lod GL_SUN_multi_draw_arrays GL_WIN_swap_hint WGL_EXT_swap_control'
[DEBUG  ] [GL          ] glShaderBinary is not available
[INFO   ] [GL          ] Backend used <glew>
[INFO   ] [GL          ] OpenGL version <b'4.6.0 - Build 26.20.100.7639'>
[INFO   ] [GL          ] OpenGL vendor <b'Intel'>
[INFO   ] [GL          ] OpenGL renderer <b'Intel(R) UHD Graphics 620'>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <b'4.60 - Build 26.20.100.7639'>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[DEBUG  ] [Shader      ] Fragment compiled successfully
[DEBUG  ] [Shader      ] Vertex compiled successfully
[DEBUG  ] [ImageSDL2   ] Load <C:\Users\abed\anaconda3\lib\site-packages\kivy\data\glsl\default.png>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[DEBUG  ] [App         ] Loading kv <C:\Users\abed\OneDrive\Documents\dev\GitHub\alant\labs\kivy\test.kv>
[DEBUG  ] [App         ] kv <C:\Users\abed\OneDrive\Documents\dev\GitHub\alant\labs\kivy\test.kv> not found
[INFO   ] [Text        ] Provider: sdl2
[DEBUG  ] [Resource    ] add <C:\WINDOWS\Fonts> in path list
[DEBUG  ] [Resource    ] add <C:\Users\abed\anaconda3\lib\site-packages\kivy\data\fonts> in path list
[DEBUG  ] [Base        ] Create provider from mouse, multitouch_on_demand
[DEBUG  ] [Base        ] Create provider from wm_touch
[DEBUG  ] [Base        ] Create provider from wm_pen
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available
[DEBUG  ] [Atlas       ] Load <C:\Users\abed\anaconda3\lib\site-packages\kivy\data\images\defaulttheme.atlas>
[DEBUG  ] [Atlas       ] Need to load 1 images
[DEBUG  ] [Atlas       ] Load <C:\Users\abed\anaconda3\lib\site-packages\kivy\data\images\defaulttheme-0.png>
[DEBUG  ] [ImageSDL2   ] Load <C:\Users\abed\anaconda3\lib\site-packages\kivy\data\images\defaulttheme-0.png>'''

**Comment by probablyfine** [comment]
@tshirtman I normally run at 150%, but I tried a few settings there and I still can't replicate the problem.

Oddly, when I bind to Window.size and then re-size the window, the callback fires repeatedly about once every second.. I'm not sure why that's happening but it makes it hard to test anything since I just get stuck in the debugger forever. Instead, I tried binding to on_resize. With that, the debugger triggered upon changing the UI scaling setting, only if the window was maximized. @abedhammoud did you get stuck in an infinite debug loop when testing this?

The only thing I really notice from your log is that you're using anaconda. I installed via pip for mine. I really don't have a good idea of if/why that would matter, but possibly worth a try with a pip install inside a virtual environment?

**Comment by matham** [comment]
Please indicate which backend matplotlib is using for each of you. Because that may well be responsible for these differences.

**Comment by probablyfine** [comment]
I'm using 'TkAgg', as reported by `matplotlib.get_backend()`

**Comment by matham** [comment]
Yeah, as I suspected, you can replicate by setting scaling to e.g. `150` and then do `environ['MPLBACKEND'] = "Qt5Agg"` before importing matplotlib. Just make sure to install `pyqt5` first.

And it's because qt/matplotlib-qt changes the DPI awareness of the application and probably the window size.

That's why I would hesitate allowing kivy to change the DPI awareness of the Kivy app since it can leads to such unintended consequences. But, we could make it users configurable if we can figure out how.

**Comment by probablyfine** [comment]
Huh, that's interesting, I wonder if anaconda comes with that backend selected by default. @abedhammoud so it sounds like you likely can change the backend to prevent this behavior if you still want to use 150% scaling. Or if for some reason that doesn't work, you could try to install matplotlib via pip and the backend should default to TkAgg I think.

> That's why I would hesitate allowing kivy to change the DPI awareness of the Kivy app since it can leads to such unintended consequences

Yeah I am starting to think this, too :/

**Comment by matham** [comment]
Please take a look at https://github.com/kivy/kivy/pull/7299. That should fix the issue.

**Comment by abedhammoud** [comment]
Amazing, thank you. I will go through all the messages above,  but I quickly wanted to comments on the back-end issue. 

@probablyfine, for me matplotlib.get_backend() returns Qt5Agg.

@matham when I try:

```python
from ctypes import windll
windll.user32.SetProcessDpiAware(-4)
import kivy
'''

I get function not found:

```python
Traceback (most recent call last):

  File "<ipython-input-5-cad2ed372c18>", line 2, in <module>
    windll.user32.SetProcessDpiAware(-4)

  File "C:\Users\abed\anaconda3\lib\ctypes\__init__.py", line 386, in __getattr__
    func = self.__getitem__(name)

  File "C:\Users\abed\anaconda3\lib\ctypes\__init__.py", line 391, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))

AttributeError: function 'SetProcessDpiAware' not found
```

do I need to install the latest kivy from master before trying this fix?

As I use anaconda, it didn't install via pip as suggested by the installation instructions, I installed kivy using conda.

a little more about my environment (which is up to date, using conda), I have the following:
```python
%conda list pywin32
# Name                    Version                   Build  Channel
pywin32                   228              py38h1e8a9f7_0    conda-forge
pywin32-ctypes            0.2.0           py38h32f6830_1002    conda-forge
```

Thank you again. I will go through the other message and see what I find.

**Comment by abedhammoud** [comment]
Thank you again,

I added the following, and the problem solved - thank you!

```python
import matplotlib
matplotlib.use('TkAgg', force=True)
'''

So it seems that the issue is specific to the Qt5Agg backend.

I truly appreciate your help - this has been bugging me for some time, and now I have the fix thanks to your pointers.
@tshirtman , @matham , @probablyfine  thank you and best wishes for the new year!

**Comment by matham** [comment]
Sorry, that's a typo on my end. It should be `SetProcessDPIAware`, with `DPI` capitalized.

**Comment by abedhammoud** [comment]
Thanks, @matham. I tried some more, I will try to summarize so hopefully this could be of help to someone.
First, it seems that the default matplotlib backend on windows and (Linux ubuntu 20.10) anaconda installation is Qt5Agg.

1. If I use matplotlib with the default backend on windows, I have the shrinking app issue when I use matplotlib to plot a graph. On Linux, the problem doesn't exist.
2. If I do the following before importing kivy as you proposed (and without changing the backend):
```python
from ctypes import windll
windll.user32.SetProcessDpiAware(-4)
import kivy
```
The application starts (but at a smaller size than step 1), however, the app doesn't shrink when I use matplotlib to plot a graph.  I will look if there is a way to ask kivy to start at a given dpi.

3. If I change the backend on windows to Tkagg before using matplotlib, then all worked well. This is the solution that I used, and my problem is resolved:
```python
import matplotlib as mpl
if kivy.metrics.platform == 'win':
    mpl.use('TkAgg', force=True)
```

Thank you. I will close this issue, I appreciate your help @matham , @tshirtman , @probablyfine
